src/SayMore/Model/ArchivingHelper.cs

```diff
@@ -66,6 +67,32 @@ internal static void ArchiveUsingIMDI(IIMDIArchivable element)
 			}
 		}
 
+		/// ------------------------------------------------------------------------------------
+		internal static void ArchiveUsingRAMP(IRAMPArchivable element)
+		{
+			// now that we added a separate title field for projects, make sure it's not empty
+			var title = string.IsNullOrEmpty(element.Title) ? element.Id : element.Title;
+
+			var model = new RampArchivingDlgViewModel(Application.ProductName, title, element.Id,
+				element.ArchiveInfoDetails, element.SetFilesToArchive, element.GetFileDescription)
+			{
+				FileCopyOverride = FileCopySpecialHandler
+			};
+			model.AppSpecificFilenameNormalization = element.CustomFilenameNormalization;
+			model.OverrideDisplayInitialSummary = fileLists => element.DisplayInitialArchiveSummary(fileLists, model);
+			model.HandleNonFatalError = (exception, s) => ErrorReport.NotifyUserOfProblem(exception, s);
+
+			element.InitializeModel(model);
+			element.SetAdditionalMetsData(model);
```

Comment:
Should this be named SetAdditionalMetaData?

Reply:
I think it defines the logic for adding data to the mets.xml file which involving in RAMP Package.

Reply:
ah, ok. :)